pennylane/devices/default_clifford.py

```diff
@@ -519,58 +698,310 @@ def _measure_state(self, tableau_simulator, circuit, global_phase):
+
+        # Partition the binary matrix to represent the subsystem
+        partition_mat = qml.math.hstack(
+            (
+                binary_mat[:, num_qubits:][:, wires],
+                binary_mat[:, :num_qubits][:, wires],
             )
-        return qml.math.array(1.0)  # // Trivial
+        )
 
-    @staticmethod
-    def _measure_expectation(tableau_simulator, meas_op, stim):
-        """Measure the expectation value with respect to the state of simulator device."""
+        # Use the reduced row echelon form for finding rank efficiently
+        # tapering always come in handy :)
+        rank = qml.math.sum(
+            qml.math.any(qml.qchem.tapering._reduced_row_echelon(partition_mat), axis=1)
+        )
 
-        # Case for simple Pauli terms
-        if isinstance(meas_op.obs, (qml.PauliZ, qml.PauliX, qml.PauliY)):
-            pauli = stim.PauliString(_GATE_OPERATIONS[meas_op.obs.name])
-            return qml.math.array(tableau_simulator.peek_observable_expectation(pauli))
-
-        # Case for simple Pauli tensor
-        if isinstance(meas_op.obs, qml.operation.Tensor):
-            expec = "".join([_GATE_OPERATIONS[name] for name in meas_op.obs.name])
-            pauli = stim.PauliString(expec)
-            return qml.math.array(tableau_simulator.peek_observable_expectation(pauli))
-
-        # Case for a Hamiltonian
-        if isinstance(meas_op.obs, qml.Hamiltonian):
-            coeffs, obs = meas_op.obs.terms()
-            expecs = qml.math.zeros_like(coeffs)
-            for idx, ob in enumerate(obs):
-                expec = "".join([_GATE_OPERATIONS[name] for name in ob.name])
-                pauli = stim.PauliString(expec)
-                expecs[idx] = tableau_simulator.peek_observable_expectation(pauli)
-            return qml.math.dot(coeffs, expecs)
-
-        # Add support for more case when the time is right
-        raise NotImplementedError(
-            f"default.clifford doesn't support expectation value calculation with {type(meas_op.obs)} at the moment."
+        # Compute the entropy
+        entropy = qml.math.log(2) * (rank - len(wires))
+
+        # Determine wether to use any log base
+        if log_base is None:
+            return entropy
+
+        return entropy / qml.math.log(log_base)
+
+    # pylint: disable=too-many-branches, too-many-statements
+    def _measure_probability(self, meas, _, **kwargs):
+        """Measure the probability of each computational basis state.
+
+        Computes the probability for each computational basis state vector. This is done by
+        building the complete basis using a vectorized integer to bit-array transformation.
+        Then, for each basis state, one iterates over each of the measured qubits, and checks
+        the possibility of that state and performs post-selection. If an impossible state is
+        encountered, the corresponding branches are tracked and marked in a visited-array.
```

Comment:
This is not clear. Having no explanation is better than having one that is not helpful.

Reply:
I have replaced this summary with the `pseudocode` for the algorithm I implement, which now breaks down the algorithm into five sequential parts. I am already specifying most of the workings, including the complexity as comments with the function definition, which is why I wanted to keep the docstring more succinct and manageable.